test/testers.jl

```diff
@@ -263,4 +271,42 @@ end
         frule_test(iterfun, (x, ẋ))
         rrule_test(iterfun, randn(), (x, x̄))
     end
+
+    @testset "unhappy path" begin
+        @testset "primal wrong" begin
+            my_identity1(x) = x
+            function ChainRulesCore.frule((_, ẏ), ::typeof(my_identity1), x)
+                return 2.5*x, ẏ
+            end
+            function ChainRulesCore.rrule(::typeof(my_identity1), x)
+                function identity_pullback(ȳ)
+                    return (NO_FIELDS, ȳ)
+                end
+                return 2.5 * x, identity_pullback
+            end
+
+            @test !isempty(metatest_get_failures(()->frule_test(my_identity1, (2.2, 3.3))))
```

Comment:
I do wonder if there's an alternative design (or just another layer of abstraction?)
so we end up with a call more like
```julia
@test_failure frule_test(my_identity1, (2.2, 3.3))
```
or
```
@test fails(() -> frule_test(my_identity1, (2.2, 3.3))
```
?

Reply:
Yeah i think the later makes sense.

Reply:
out of interest, why not the macro (that just wraps the call in an anonymous function and passes it to `fails`)?

Reply:
the macro looks like `@test_throws` and `@test_broken`
but it does not at all work like those do.

`@test_throws` catchs the exception.
`@test_broken x` is very similar to `@test !x` + some extra stuff

I would worry that people might think  `@test_failure x`  is the same as same as `@test !x`

Reply:
> people might think @test_failure x is the same as same as @test !x

yeah, great point.